pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -1176,84 +1188,1309 @@ func Test_server_Insert(t *testing.T) {
+				args: args{
+					ctx: ctx,
+					req: &payload.Insert_Request{
+						Vector: &payload.Object_Vector{
+							Id:     id,
+							Vector: []float32{},
+						},
+					},
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    dim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Float.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				want: want{
+					// IncompatibleDimensionSize error will be returned
+					err: func() error {
+						err := errors.ErrIncompatibleDimensionSize(0, 3)
+						return status.WrapWithInvalidArgument("Insert API Incompatible Dimension Size detected",
+							err,
+							&errdetails.RequestInfo{
+								RequestId:   "1",
+								ServingData: errdetails.Serialize(nil),
+							},
+							&errdetails.BadRequest{
+								FieldViolations: []*errdetails.BadRequestFieldViolation{
+									{
+										Field:       "vector dimension size",
+										Description: err.Error(),
+									},
+								},
+							},
+							&errdetails.ResourceInfo{
+								ResourceType: ngtResourceType + "/ngt.Insert",
+								ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+							})
+					}(),
+				},
+			}
+		}(),
+
+		// Decision Table Testing
+		func() test {
+			vec2 := []float32{3, 2, 1} // used in beforeFunc
+			req := &payload.Insert_Request{
+				Vector: &payload.Object_Vector{
+					Id:     id,
+					Vector: intVec,
+				},
+				Config: &payload.Insert_Config{
+					SkipStrictExistCheck: false,
+				},
+			}
+
+			return test{
+				name: "Decision Table Testing case 1.1: Insert duplicated request fail when SkipStrictExistCheck is false (duplicated ID)",
+				args: args{
+					ctx: ctx,
+					req: req,
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    intVecDim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Uint8.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				beforeFunc: func(s *server) {
+					s.ngt.Insert(id, vec2)
+				},
+				want: want{
+					err: status.WrapWithAlreadyExists(fmt.Sprintf("Insert API uuid %s already exists", id), errors.ErrUUIDAlreadyExists(id),
+						&errdetails.RequestInfo{
+							RequestId:   req.GetVector().GetId(),
+							ServingData: errdetails.Serialize(req),
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: ngtResourceType + "/ngt.Insert",
+							ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+						}),
+				},
+			}
+		}(),
+		func() test {
+			id2 := "uuid2"             // use in beforeFunc
+			vec2 := []float32{3, 2, 1} // use in beforeFunc
```

Comment:
could you use generator instead of it?

Reply:
same context of https://github.com/vdaas/vald/pull/1552#discussion_r832758479